packages/composites/set-token-index/src/endpoint/allocations.ts

```diff
@@ -95,10 +96,19 @@ export const execute: ExecuteWithConfig<Config> = async (input, context, config)
     // Token balances are coming already normalized as 18 decimals token
     const allocations = await Promise.all(
       addresses.map(async (address: string, i: number) => {
-        const token = await getToken(context, jobRunID, address)
+        const token = (await getToken(
+          context,
+          jobRunID,
+          address,
+        )) as unknown as TA.types.TokenAllocation
+
         return {
-          balance: balances[i].toString(),
-          ...token,
+          balance: new BigNumber(balances[i].toString())
+            .shiftedBy(-(18 - token.decimals))
```

Comment:
Can we also handle tokens with greater than 18 decimals? Not sure how common it is, but found at least one example: https://etherscan.io/token/0xaba8cac6866b83ae4eec97dd07ed254282f6ad8a#readContract

Reply:
This should correctly handle tokens with decimals greater than 18 decimals, as in that case it will scale up rather than down, by -(18-tokenDecimals), tokenDecimals=24, which would give +6. So it would scale the provided 18 decimal number up to the correct 24 decimal value.

Reply:
Ahh yep that makes sense, nice!